src/realm.h

```diff
@@ -1274,6 +1274,14 @@ RLM_API bool realm_get_class_keys(const realm_t*, realm_class_key_t* out_keys, s
  */
 RLM_API bool realm_find_class(const realm_t*, const char* name, bool* out_found, realm_class_info_t* out_class_info);
 
+/**
+ * Find class info based on a realm object.
+ *
+ * @param object object which is an instance of the class.
+ * @return a `realm_class_info_t` that will contain information about the class.
+ */
+RLM_API realm_class_info_t realm_object_get_class(realm_object_t* object) RLM_API_NOEXCEPT;
```

Comment:
Return a `ptr` to `realm_class_info_t`, returning by value, is expensive. I suggest changing the signature into:
`RLM_API realm_class_info_t* realm_object_get_class(realm_object_t* object) RLM_API_NOEXCEPT;`

Reply:
I don't think we gain anything by placing a copy of `realm_class_info_t` on the heap rather than on the stack. Returning a pointer adds an extra step where we need to free it later.

Reply:
I don't think copying the whole struct and putting it into the stack  is the way to go. I propose to follow the same pattern of:
`RLM_API bool realm_find_class(const realm_t*, const char* name, bool* out_found, realm_class_info_t* out_class_info);`

No extra deletion for the SDK, but at the same time no heavy copy.